tests/capture/test_capture_cond.py

```diff
@@ -516,3 +611,88 @@ def test_circuit_consts(self, pred, arg, expected):
         jaxpr = jax.make_jaxpr(circuit_with_consts)(*args)
         res_ev_jxpr = jax.core.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, *args)
         assert np.allclose(res_ev_jxpr, expected), f"Expected {expected}, but got {res_ev_jxpr}"
+
+    @pytest.mark.parametrize("reset", [True, False])
+    @pytest.mark.parametrize("postselect", [None, 0, 1])
+    @pytest.mark.parametrize("shots", [None, 20])
+    def test_mcm_predicate_execution(self, reset, postselect, shots):
+        """Test that QNodes executed with mid-circuit measurement predicates for
+        qml.cond give correct results."""
+        device = qml.device("default.qubit", wires=3, shots=shots, seed=jax.random.PRNGKey(1234))
+
+        def true_fn(arg):
+            qml.RX(arg, 0)
+
+        def false_fn(arg):
+            qml.RY(arg, 0)
+
+        @qml.qnode(device)
+        def f(x, y):
+            qml.RX(x, 0)
+            m = qml.measure(0, reset=reset, postselect=postselect)
+
+            qml.cond(m, true_fn, false_fn)(y)
+            return qml.expval(qml.Z(0))
```

Comment:
I _think_ the predicate here will not affect the output value of the QNode, because RX and RY rotations are equivalent if we start in the |0> or |1> state and measure Z afterwards.
Could change the size of the argument in addition:
```suggestion
        def true_fn(arg):
            qml.RX(arg, 0)

        def false_fn(arg):
            qml.RY(3 * arg, 0)

        @qml.qnode(device)
        def f(x, y):
            qml.RX(x, 0)
            m = qml.measure(0, reset=reset, postselect=postselect)

            qml.cond(m, true_fn, false_fn)(y)
            return qml.expval(qml.Z(0))
```